Classes/Player/Tracks/PKCaptionsAssetResourceLoaderDelegate.swift

```diff
@@ -0,0 +1,207 @@
+             a different choice.  This attribute is OPTIONAL.  Its absence
+             indicates an implicit value of NO.
+             */
+            if subtitle.isDefault {
+                string.append(",DEFAULT=YES")
+            }
+            
+            /*
+             The value is an enumerated-string; valid strings are YES and NO.
+             This attribute is OPTIONAL.  Its absence indicates an implicit
+             value of NO.  If the value is YES, then the client MAY choose to
+             play this Rendition in the absence of explicit user preference
+             because it matches the current playback environment, such as
+             chosen system language.
+             
+             If the AUTOSELECT attribute is present, its value MUST be YES if
+             the value of the DEFAULT attribute is YES.
+             */
+            if subtitle.autoSelect {
+                string.append(",AUTOSELECT=YES")
+            }
+            
+            /*
+             The value is an enumerated-string; valid strings are YES and NO.
+             This attribute is OPTIONAL.  Its absence indicates an implicit
+             value of NO.  The FORCED attribute MUST NOT be present unless the
+             TYPE is SUBTITLES.
+             */
+            if subtitle.forced {
+                string.append(",FORCED=YES")
+            }
+            
+            /*
+             The value is a quoted-string containing one of the standard Tags
+             for Identifying Languages [RFC5646], which identifies the primary
+             language used in the Rendition.  This attribute is OPTIONAL.
+             */
+            if !subtitle.language.isEmpty {
+                string.append("""
+                    ,LANGUAGE="\(subtitle.language)"
+                    """)
+            }
+            
+            /*
+             The value is a quoted-string containing one or more Uniform Type
+             Identifiers [UTI] separated by comma (,) characters.  This
+             attribute is OPTIONAL.  Each UTI indicates an individual
+             characteristic of the Rendition.
+             
+             A SUBTITLES Rendition MAY include the following characteristics:
+             "public.accessibility.transcribes-spoken-dialog",
+             "public.accessibility.describes-music-and-sound", and
+             "public.easy-to-read" (which indicates that the subtitles have
+             been edited for ease of reading).
+             */
+            if !subtitle.characteristics.isEmpty {
+                string.append("""
+                    ,CHARACTERISTICS="\(subtitle.characteristics)"
+                    """)
+            }
+            
+            string.append("\n")
+            
+            allSubtitles.append(string)
+        }
+        return allSubtitles
+    }
+    
+    func getSubtitlem3u8(forSubtitle subtitle: PKExternalSubtitle) -> String {
```

Comment:
Consider moving this function to PKExternalSubtitle (as `buildM3u8Playlist()` or similar).

Reply:
Yes, you'r right, better approach. Will change it.